### PR TITLE
Separate timestamp related test from db_basic_test

### PR DESCRIPTION
**Pull request by zhichao-cao**
In some of the test, db_basic_test may cause time out due to its long running time. Separate the timestamp related test from db_basic_test to avoid the potential issue.

Test plan: pass make asan_check

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,6 +34,7 @@ manifest_dump
 sst_dump
 blob_dump
 block_cache_trace_analyzer
+db_basic_test_with_timestamp
```
Comment (riversand963) [inline .gitignore]: You can follow the naming convention for unit test here. Renaming it to `db_with_timestamp_basic_test` will make it unnecessary for this line.

File: db/db_basic_test_with_timestamp.cc [hunk 2]
```diff
@@ -0,0 +1,808 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#include "db/db_test_util.h"
```
Comment (riversand963) [inline db/db_basic_test_with_timestamp.cc]: One empty line between first include and copy right.

**Comment by zhichao-cao** [comment]
> LGTM with a few minor comments.
> Thanks @zhichao-cao for fixing it.

Thanks for the reviews. I will change accordingly.

**Comment by pdillinger** [comment]
How did this get landed? It looks like it broke Travis and AppVeyor builds, even at development time: https://travis-ci.org/github/facebook/rocksdb/builds/661778814 https://ci.appveyor.com/project/Facebook/rocksdb/builds/31437574

**Pull request closed without merging**